src/words/fast.jl

```diff
@@ -95,6 +95,22 @@ a word boundary should follow `s`.
     false
 """
 function lookahead(ts::TokenBuffer, s; boundary = false)
+  if typeof(s) == Regex
+    tmp_str = ""
+    for i in ts.idx : length(ts.input)
+      tmp_str *= ts.input[i]
```

Comment:
@oxinabox using `replaces()` we would still need to do a Regex search in the buffer.
Here, for each regex I have done a matching operation each time after appending a character to `tmp_str`, which makes the time complexity `O(n*n*m)`
(where n is length of rest of buffer to be processed and m is length of tokens given to `atoms`.)
If we make a slight change and for each regex(given to `atoms`) first extract all of the buffer and then do matching operation once, the time complexity will reduce to `O(n*m)`.